lib/ansible/playbook/role/__init__.py

```diff
@@ -442,6 +442,13 @@ def add_parent(self, parent_role):
     def get_parents(self):
         return self._parents
 
+    def get_dep_chain(self):
+        dep_chain = []
+        for parent in self._parents:
+            dep_chain.extend(parent.get_dep_chain())
+            dep_chain.append(parent)
+        return dep_chain
+
```

Comment:
Posting for my benefit, this was my PoC fix to the last attempt, although it was in `role_include.py`:

```diff
diff --git a/lib/ansible/playbook/base.py b/lib/ansible/playbook/base.py
index 564ce0cd3a..c8fb957132 100644
--- a/lib/ansible/playbook/base.py
+++ b/lib/ansible/playbook/base.py
@@ -871,7 +871,10 @@ class Base(FieldAttributeBase):
     def get_dep_chain(self):

         if hasattr(self, '_parent') and self._parent:
-            return self._parent.get_dep_chain()
+            chain = self._parent.get_dep_chain() or []
+            if parent_role := getattr(self, '_parent_role', None):
+                chain.append(parent_role)
+            return chain
         else:
             return None
```

I like using `self._parents` here, and this does effectively implement the same change.